### DISCUSSION TITLE
FFID score usage

### DISCUSSION THREAD

**Discussion by jpfeuffer** [OpenMS/OpenMS#5129] [Code]
Hi @OpenMS/core,

specifically @hendrikweisser.
I was recently spending some time in FFID during my efforts to optimize ProteomicsLFQ (or LFQ pipelines) in general.
I found the usage of some DIA scores a bit doubtful and would like to know what kind of benefits were expected from that scores.
I specifically mean the _isotope scores_:
What is isotope_corr and isotope_overlap doing for us with the "fake" transitions of FFID?
If I understand it correctly, it adds isotopes to each transition and scores the intensities with the expected ratios. BUT: our (product) transitions are already the isotopes, right? So, I did not yet see the benefit from that score. I would actually consider the dotprod/manhattan score as more informative, since it goes over all transitions together and compares them to an isotope profile (although I am relatively sure that it is a bit bugged and does not work as intended with charges =! 1). This score is on the other hand not yet used but only calculated.

Related to usage/non-usage: Wouldn't it make sense to disable the scores that are not even used, so that they are not calculated in the first place and we save time?

A general question regarding the usage of those scores: Are they even used when you only have "internal" features (i.e. the SVM is not running)?

**Comment by hendrikweisser** [comment]
I faintly recall that I discussed the different scores with Hannes at the time, but I don't remember the isotope scores. Were they added after FFId was developed? In any case, I can't comment on how they work, so it may well be that they are not suitable for FFId scoring. And yes, it would make sense to me to disable the calculation of scores that aren't useful.

With only "internal" IDs (no SVM), the DIA scores aren't used in FFId itself, but they may contribute to the overall OpenSWATH score that is used as the feature quality score.

**Reply by jpfeuffer** [reply]
Thanks a lot. Then I investigate the scores a bit more and test their potential removal/deactivation.
Is the overall feature quality used for some internal filtering already? Or only for manual Post-Processing after FFiD?

Btw: I don't know for sure if those scores were added later. I would however tend to "no" since I think they were specifically listed in some parameter of FFID.

**Reply by hendrikweisser** [reply]
> Is the overall feature quality used for some internal filtering already?

No, shouldn't be.

**Reply by hroest** [reply]
I think the scores are already there, but you are right @jpfeuffer that these scores assume that you are dealing with a monoisotopic peak and it then looks for isotopes higher and lower. So this only makes sense for a monoisotopic peak
